nifi-extension-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/excel/SplitExcel.java

```diff
@@ -0,0 +1,239 @@
+                        }
+
+                        if (!originalRows.isEmpty()) {
+                            newSheet.copyRows(originalRows, originalSheet.getFirstRowNum(), CELL_COPY_POLICY);
+                        }
+
+                        FlowFile newFlowFile = session.create(originalFlowFile);
+                        try (final OutputStream out = session.write(newFlowFile)) {
+                            newWorkbook.write(out);
+                            workbookSplits.add(new WorkbookSplit(index, newFlowFile, originalSheetName, originalRows.size()));
+                        }
+                    }
+
+                    index++;
+                }
+            });
+        } catch (ExcelRuntimeException | ProcessException e) {
+            getLogger().error("Failed to split {}", originalFlowFile, e);
+            session.remove(workbookSplits.stream()
+                    .map(WorkbookSplit::content)
+                    .toList());
+            workbookSplits.clear();
+            session.transfer(originalFlowFile, REL_FAILURE);
+            return;
+        }
+
+        final String fragmentId = UUID.randomUUID().toString();
+        final String originalFileName = originalFlowFile.getAttribute(CoreAttributes.FILENAME.key());
+        final int extensionIndex = originalFileName.lastIndexOf(".");
+        String originalFileNameWithoutExtension = originalFileName;
+        String originalFileNameExtension = "";
+
+        if (extensionIndex > -1) {
+            originalFileNameWithoutExtension = originalFileName.substring(0, extensionIndex);
+            originalFileNameExtension = originalFileName.substring(extensionIndex);
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(FRAGMENT_COUNT, String.valueOf(workbookSplits.size()));
+        attributes.put(FRAGMENT_ID, fragmentId);
+        attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFileName);
+
+        for (WorkbookSplit split : workbookSplits) {
+            attributes.put(CoreAttributes.FILENAME.key(), originalFileNameWithoutExtension + "-" + split.index() + originalFileNameExtension);
+            attributes.put(FRAGMENT_INDEX, Integer.toString(split.index()));
+            attributes.put(SHEET_NAME, split.sheetName());
+            attributes.put(TOTAL_ROWS, Integer.toString(split.numRows()));
+            session.putAllAttributes(split.content(), attributes);
+        }
+
+        session.transfer(originalFlowFile, REL_ORIGINAL);
+        final List<FlowFile> flowFileSplits = workbookSplits.stream()
+                .map(WorkbookSplit::content)
+                .toList();
+
+        if (flowFileSplits.size() > 1) {
+            session.transfer(flowFileSplits, REL_SPLIT);
+        } else {
+            getLogger().warn("Original file {} did not have more than one sheet, nothing to split", originalFlowFile);
+            session.remove(flowFileSplits);
+        }
```

Comment:
I'm not sure this check is necessary, I think it is better to always write the split files, even if there is only one, and avoid the warning.